src/components/code-demo/code-demo.styles.ts

```diff
@@ -0,0 +1,199 @@
+    --preview-backdrop: var(--wa-color-surface-lowered, rgb(0 0 0 / 0.25));
+    --preview-resize: inline;
+    --preview-min-width: min-content;
+    --preview-max-width: 100%;
+    --preview-padding: var(--wa-space-2xl, 2rem);
+    --divider-width: var(--wa-border-width-s, 1px);
+    --viewport-initial-aspect-ratio: 16 / 9;
+    --viewport-bezel-width: 0.25em;
+
+    --code-expand-duration: var(--wa-transition-fast, 0.3s);
+    --code-collapse-duration: var(--wa-transition-normal, 0.3s);
+
+    display: flex;
+    flex-flow: column;
+    border: var(--wa-border-style) var(--wa-panel-border-width) var(--wa-color-neutral-border-quiet);
+    border-radius: var(--wa-code-demo-rounding, var(--wa-border-radius-m));
+    color: var(--wa-color-text-normal);
+    margin-block-end: var(--wa-flow-spacing);
+    background: var(--preview-backdrop);
+    interpolate-size: allow-keywords;
+  }
+
+  /* Different defaults for isolated demos */
+  :host([viewport]) {
+    --preview-resize: both;
+    --preview-backdrop: var(--preview-background);
+    --preview-padding: var(--wa-space-l, 1rem);
+  }
+
+  #preview {
+    display: block;
+    padding: var(--preview-padding);
+    border-block-end: inherit;
+    border-block-end-width: var(--divider-width);
+    border-start-start-radius: inherit;
+    border-start-end-radius: inherit;
+    background: var(--preview-background);
+    contain: inline-size;
+    container-type: inline-size;
+    container-name: preview;
+
+    &:has(#viewport) {
+      background: var(--preview-backdrop);
+    }
+
+    &:not(:has(#viewport)) {
+      max-width: min(var(--preview-max-width), 100%);
+      min-width: var(--preview-min-width, min-content);
+    }
+
+    &:not(:has(#viewport)),
+    #viewport {
+      resize: var(--preview-resize);
+      overflow: auto;
+    }
+
+    > :first-child {
+      margin-block-start: 0;
+    }
+
+    > :last-child {
+      margin-block-end: 0;
+    }
+  }
+
+  #source {
+    border-block-end: inherit;
+    overflow: hidden;
+    transition-property: height, display;
+    transition-behavior: allow-discrete;
+
+    &::slotted(pre) {
+      position: relative;
+      border-radius: 0 !important;
+      margin: 0;
+      white-space: normal;
+    }
+
+    &:has(+ #buttons) {
+      border-end-start-radius: 0;
+      border-end-end-radius: 0;
+    }
+
+    &:not(:has(+ #buttons)) {
+      border-bottom: none;
+    }
+  }
+
+  [part~='toggle'] wa-icon {
+    transition-property: rotate;
+  }
+
+  :host(:not([open])) {
+    #source,
+    [part~='toggle'] wa-icon {
+      transition-duration: var(--code-collapse-duration);
+    }
+
+    #source {
+      /* Collapsed */
+      height: 0px;
+      display: none;
+    }
+  }
+
+  :host([open]) {
+    #source,
+    [part~='toggle'] wa-icon {
+      transition-duration: var(--code-expand-duration);
+    }
+
+    #source {
+      /* Expanded */
+      height: auto;
+      display: block;
+    }
```

Comment:
This transitions the height on expand:

```suggestion
    #source {
      /* Expanded */
      height: auto;
      display: block;
    }

    @starting-style {
      #source {
        height: 0px;
        display: block;
      }
    }
```

Reply:
Have you tried it? I'd swear it was among the many things I tried that didn't work…

Reply:
Yes, it works in Chrome. My first go was a fail because I forgot to add the `#source` selector inside `@starting-style` but it works with the above.